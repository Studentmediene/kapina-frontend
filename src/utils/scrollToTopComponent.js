import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

export class ScrollToTop extends React.Component {
  static propTypes = {
    children: PropTypes.array.isRequired,
    location: PropTypes.object.isRequired,
  };
  componentDidUpdate(prevProps) {
    if (this.props.location !== prevProps.location) {
      window.scrollTo(0, 0);
    }
  }

  render() {
    return this.props.children;
  }
}

export default withRouter(ScrollToTop);
