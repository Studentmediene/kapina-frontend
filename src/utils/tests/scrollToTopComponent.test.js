import React from 'react';
import { shallow } from 'enzyme';

import { ScrollToTop } from '../scrollToTopComponent';

describe('<ScrollToTop />', () => {
  it('renders correctly', () => {
    const props = {
      children: [<div key={1} />],
      location: {},
    };

    const tree = shallow(<ScrollToTop {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
