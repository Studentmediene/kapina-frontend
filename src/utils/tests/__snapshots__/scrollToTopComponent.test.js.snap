// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollToTop /> renders correctly 1`] = `
Array [
  <div
    key="1"
  />,
]
`;
